src/terms/hubbard.jl

```diff
@@ -0,0 +1,305 @@
+            # Find the index of the projector that matches the manifold label
+            proj_index = findfirst(p -> p.iatom == orb.iatom && p.species == orb.species &&
+                                       p.n == orb.n && p.l == orb.l && p.m == orb.m, labels)
+            if proj_index !== nothing
+                manifold_projectors[ik][:, iproj] = projk[:, proj_index]
+            else
+                @warn "Projector for $(orb.species) with n=$(orb.n), l=$(orb.l), m=$(orb.m)
+                       not found in projectors."
+            end
+        end
+    end
+    return (; manifold_labels, manifold_projectors)
+end
+
+function compute_overlap_matrix(basis::PlaneWaveBasis{T};
+                                manifold  = nothing,
+                                positions = basis.model.positions) where {T}
+    proj = atomic_orbital_projectors(basis; manifold, positions)
+    projectors = proj.projectors
+    labels = proj.labels
+    overlap_matrix = Vector{Matrix{T}}(undef, length(basis.kpoints))
+
+    for (ik, projk) in enumerate(projectors)
+        overlap_matrix[ik] = abs2.(projk' * projk)
+    end
+
+    return (; overlap_matrix, projectors, labels)
+end
+
+"""
+Symmetrize the Hubbard occupation matrix according to the l quantum number of the manifold.
+"""
+function symmetrize_nhub(n_IJ::Array{Matrix{Complex{T}}}, lattice, symmetry, positions) where {T}
+    # For now we apply symmetries only on nII terms, not on cross-atom terms (nIJ)
+    # WARNING: To implement +V this will need to be changed!
+
+    nspins = size(n_IJ, 1)
+    natoms = size(n_IJ, 2)
+    nsym = length(symmetry)
+    l = Int64((size(n_IJ[1, 1, 1], 1)-1)/2)
+    WigD = Wigner_sym(l, lattice, symmetry)
+
+     # Initialize the n_IJ matrix
+    ns = Array{Matrix{Complex{T}}}(undef, nspins, natoms, natoms)
+    for σ in 1:nspins, iatom in 1:natoms, jatom in 1:natoms
+        ns[σ, iatom, jatom] = zeros(Complex{T},
+                                    size(n_IJ[σ, iatom, jatom], 1),
+                                    size(n_IJ[σ, iatom, jatom], 2))
+    end
+
+    for σ in 1:nspins, iatom in 1:natoms, isym in 1:nsym
+        for m1 in 1:size(ns[σ, iatom, iatom], 1), m2 in 1:size(ns[σ, iatom, iatom], 2)
+            sym_atom = find_symmetric(iatom, symmetry, isym, positions)
+            # TODO: Here QE flips spin for time-reversal in collinear systems, should we?
+            for m0 in 1:size(n_IJ[σ, iatom, iatom], 1), m00 in 1:size(n_IJ[σ, iatom, iatom], 2)
+                ns[σ, iatom, iatom][m1, m2] += WigD[m0, m1, isym] *
+                                               n_IJ[σ, sym_atom, sym_atom][m0, m00] *
+                                               WigD[m00, m2, isym]
+            end
+        end
+    end
+    ns .= ns / nsym
+
+    return ns
+end
+
+"""
+Find the symmetric atom index for a given atom and symmetry operation
+"""
+function find_symmetric(iatom::Int64, symmetry::Vector{SymOp{T}},
```

Comment:
It looks like this should not exist, and instead `find_symmetry_preimage` should be called.